event.go

```diff
@@ -121,9 +117,67 @@ func (ch *Channel) SendEvent(event *Event, userID string) error {
 
 	event.User = &User{ID: userID}
 
-	req := eventRequest{Event: event}
+	req := struct {
+		Event *Event `json:"event"`
+	}{
+		Event: event,
+	}
 
 	p := path.Join("channels", url.PathEscape(ch.Type), url.PathEscape(ch.ID), "event")
 
 	return ch.client.makeRequest(http.MethodPost, p, nil, req, nil)
 }
+
+// UserCustomEvent is a custom event sent to a particular user.
+type UserCustomEvent struct {
+	// Type should be a custom type. Using a built-in event is not supported here.
+	Type string `json:"type"`
+
+	ExtraData map[string]interface{} `json:"-"`
+
+	CreatedAt time.Time `json:"created_at,omitempty"`
+}
+
+type userCustomEventForJSON UserCustomEvent
+
+func (e *UserCustomEvent) UnmarshalJSON(data []byte) error {
+	// TODO: merge this method with Event.UnmarshalJSON
```

Comment:
would be nice if we could address this one too

Reply:
however, I would go that way since they are quite different. That's why just dropping this todo is good addressing to me.